### ISSUE TITLE
`kron(a::RowVector, b::RowVector)` should return RowVector

### ISSUE THREAD

**Issue by i-apellaniz** [JuliaLang/julia#22131]
I think it is reasonable to implement this into the `kron` function.

At the time of reporting this issue, 
```julia
julia> kron([1,2]',[3,4]')
1×4 Array{Int64,2}:
 3  4  6  8
```
which is reasonable, but not desirable in my opinion.

I think it should return
```julia
julia> kron([1,2]',[3,4]')
1×4 RowVector{Int64,Array{Int64,1}}:
 3  4  6  8
```

**Comment by andyferris** [comment]
It should. Unfortunately, there is no good generic interface for making new row vectors of a different size, *a la* `similar`. 

I'd propose a special index type for the first dimension of `RowVector`, maybe combined with using the indices (not size) in `similar`. Alternatively, we could find and patch as many common cases like this one where this goes wrong...

**Comment by i-apellaniz** [comment]
What about this patch?
```julia
kron(a::RowVector, b::RowVector) = kron(a',b')'
```

I think it is not the best option, but is at least clean enough (and maybe fast enough) to implement it until a new solution is found.

**Comment by dkarrasch** [comment]
This has been fixed in #35755 (v1.5).
